### PR TITLE
Automigrations: Create addon-postcss automigration

### LINKED ISSUE #26129: [Bug]: @storybook/addon-postcss doesn't exist in Storybook 8
A lot of Storybook 6 users are using https://github.com/storybookjs/addon-postcss, which is deprecated. The docs of @storybook/addon-postcss are pointing to @storybook/addon-styling, which is also deprecated. So the recommendation nowadays is to use @storybook/addon-styling-webpack.

We want to notify users during the automigration process that they should migrate away from `@storybook/addon-postcss` to `@storybook/addon-styling-webpack`.

## Acceptance criteria

- If the user has `@storybook/addon-postcss` installed
  - An automigration should warn the user that the plugin is deprecated and is not working with Storybook 8. The user should migrate over to `@storybook/addon-styling-webpack`

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Closes https://github.com/storybookjs/storybook/issues/26129

## What I did

I have added an automigration to detect whether @storybook/addon-postcss is installed. If so, the user should be warned and notified to rather install another addon.

### Testing

- Checkout a Storybook 6 project
- Run 

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

1. Checkout a Storybook 6 project
2. Run `storybook upgrade`
3. If it has the `@storybook/addon-postcss` setup as an addon in `.storybook/main.js`, the automigration message should appear.

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This pull request has been released as version `0.0.0-pr-26228-sha-c671ac97`. Try it out in a new sandbox by running `npx storybook@0.0.0-pr-26228-sha-c671ac97 sandbox` or in an existing project with `npx storybook@0.0.0-pr-26228-sha-c671ac97 upgrade`.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-26228-sha-c671ac97`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-26228-sha-c671ac97) |
| **Triggered by** | @valentinpalkovic |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`valentin/create-automigration-for-postcss-addon`](https://github.com/storybookjs/storybook/tree/valentin/create-automigration-for-postcss-addon) |
| **Commit** | [`c671ac97`](https://github.com/storybookjs/storybook/commit/c671ac976738a23cf0059d8f42f195f41bb6a5e8) |
| **Datetime** | Wed Feb 28 10:28:42 UTC 2024 (`1709116122`) |
| **Workflow run** | [8078902825](https://github.com/storybookjs/storybook/actions/runs/8078902825) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=26228`_
</details>

### CODE REVIEW

File: code/lib/cli/src/automigrate/fixes/addon-postcss.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    return { hasAddonPostcss: true };
+  },
+
+  prompt() {
+    return dedent`
+      ${chalk.bold(
+        'Attention'
+      )}: We've detected that you're using the following package which might be incompatible since Storybook 8:
```
Comment (JReinhold) [inline code/lib/cli/src/automigrate/fixes/addon-postcss.ts:32]:
```suggestion
      )}: We've detected that you're using the following package which are incompatible with Storybook 8 and beyond:
```

File: code/lib/cli/src/automigrate/fixes/addon-postcss.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+      - ${chalk.cyan(`@storybook/addon-postcss`)}
+      
+      Please migrate to ${chalk.cyan(
+        `@storybook/addon-styling-webpack`
+      )} once the migration is complete.
```
Comment (JReinhold) [inline code/lib/cli/src/automigrate/fixes/addon-postcss.ts:38]:
```suggestion
      )} once you're done upgrading.
```

File: code/lib/cli/src/automigrate/fixes/addon-postcss.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  prompt() {
+    return dedent`
+      ${chalk.bold(
+        'Attention'
+      )}: We've detected that you're using the following package which might be incompatible since Storybook 8:
+
+      - ${chalk.cyan(`@storybook/addon-postcss`)}
+      
+      Please migrate to ${chalk.cyan(
+        `@storybook/addon-styling-webpack`
```
Comment (JReinhold) [inline code/lib/cli/src/automigrate/fixes/addon-postcss.ts:37]: What about Vite projects? Are we just hoping that they wouldn't have `@storybook/addon-postcss`?

Comment (valentinpalkovic) [reply code/lib/cli/src/automigrate/fixes/addon-postcss.ts:37]: They were not using the addon. It was a webpack exclusive one: https://github.com/storybookjs/addon-postcss/blob/master/src/index.ts

**Review by JReinhold — APPROVED** [review]
✨

**Pull request merged**